plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -288,14 +299,74 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
+        if (type instanceof TimestampType) {
+            if (storageFormat == AVRO) {
+                // TODO https://github.com/trinodb/trino/issues/20481
+                throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp type with %s format is not supported.".formatted(storageFormat));
```

Comment:
`AVRO` is an enum value. It should be `Avro` from the user-perspective in strict. 
```suggestion
                throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp type with Avro format is not supported.");
```